src/vs/workbench/browser/parts/editor/editorGroupsControl.ts

```diff
@@ -969,6 +1015,31 @@ export class EditorGroupsControl extends Themable implements IEditorGroupsContro
 		// Silo Three
 		this.silos[Position.THREE] = $(this.parent).div({ class: 'one-editor-silo editor-three' });
 
+		// Center Layout stuff
```

Comment:
Minor. Feels like code duplication, we do the creation and listeners twice. Can't we have a helper method that just takes a centered sash and nicely initilizes it. For that the DragStart and Drag methods would have to take sashes as parameters and based on Sash do different things.
Which might not be too bad, currently I feel like there are too many methods.

Reply:
I don't think that's a good idea... that would make the individual method body larger... and break the coding style for sashes... I mean the class is pretty big but that would require an additional PR that really breaks the class apart. I think it's much more clean the way it is... what we could do is outfactor the initialization for the sashes. I did that in this commit https://github.com/SrTobi-Forks/vscode/commit/abccb7738d4803aa4241ec9b5dab10162cbd5602 (it's not part of this PR at the moment) but I am not really sure if that makes the code more structured (in fact, it adds one more method :smile: ).

Reply:
Ok, makes sense